resolve_test.go

```diff
@@ -234,3 +235,87 @@ func TestBadDomain(t *testing.T) {
 		t.Error("expected malformed address to fail to parse")
 	}
 }
+
+func TestCustomResolver(t *testing.T) {
+	ip1 := net.IPAddr{IP: net.ParseIP("1.2.3.4")}
+	ip2 := net.IPAddr{IP: net.ParseIP("2.3.4.5")}
+	ip3 := net.IPAddr{IP: net.ParseIP("3.4.5.6")}
+	ip4 := net.IPAddr{IP: net.ParseIP("4.5.6.8")}
+	ip5 := net.IPAddr{IP: net.ParseIP("5.6.8.9")}
+	def := &MockResolver{
+		IP: map[string][]net.IPAddr{
+			"example.com": []net.IPAddr{ip1},
+		},
+	}
+	custom1 := &MockResolver{
+		IP: map[string][]net.IPAddr{
+			"custom.test":         []net.IPAddr{ip2},
+			"another.custom.test": []net.IPAddr{ip3},
```

Comment:
maybe add `"more.custom.test" : []net.IPAddr{ip3}` (or anything other than `ip4`) here. It will demonstrate that even if the more generic resolver has a record that it won't be used instead of the more specific resolver

Reply:
good idea, will add.

Reply:
done.